src/xml/XMLTag.cpp

```diff
@@ -101,36 +101,56 @@ bool XMLTag::hasAttribute(const std::string &attributeName)
   return utils::contained(attributeName, _attributes);
 }
 
-double XMLTag::getDoubleAttributeValue(const std::string &name) const
+double XMLTag::getDoubleAttributeValue(const std::string &name, std::optional<double> default_value) const
 {
   std::map<std::string, XMLAttribute<double>>::const_iterator iter;
   iter = _doubleAttributes.find(name);
-  PRECICE_ASSERT(iter != _doubleAttributes.end());
-  return iter->second.getValue();
+  if (iter != _doubleAttributes.end()) {
+    return iter->second.getValue();
+  }
+  if (default_value) {
+    return default_value.value();
+  }
+  PRECICE_ERROR("Not found.");
 }
 
-int XMLTag::getIntAttributeValue(const std::string &name) const
+int XMLTag::getIntAttributeValue(const std::string &name, std::optional<int> default_value) const
 {
   std::map<std::string, XMLAttribute<int>>::const_iterator iter;
   iter = _intAttributes.find(name);
-  PRECICE_ASSERT(iter != _intAttributes.end());
-  return iter->second.getValue();
+  if (iter != _intAttributes.end()) {
+    return iter->second.getValue();
+  }
+  if (default_value) {
+    return default_value.value();
+  }
+  PRECICE_ERROR("Not found.");
 }
 
-const std::string &XMLTag::getStringAttributeValue(const std::string &name) const
+const std::string &XMLTag::getStringAttributeValue(const std::string &name, std::optional<std::string> default_value) const
 {
   std::map<std::string, XMLAttribute<std::string>>::const_iterator iter;
   iter = _stringAttributes.find(name);
-  PRECICE_ASSERT(iter != _stringAttributes.end(), name);
-  return iter->second.getValue();
+  if (iter != _stringAttributes.end()) {
+    return iter->second.getValue();
+  }
+  if (default_value) {
+    return default_value.value();
+  }
+  PRECICE_ERROR("Not found.");
```

Comment:
The attributes directly depend on the existing configuration. Attributes with defaults will be always be in the map, while missing attributes without defaults will raise an error in the parsing stage. Meaning a user cannot trigger this.

Hence, the check should be an assertion:

```suggestion
  PRECICE_ASSERT(default_value, "The XMLAttribute doesn't exist, check its default.");
  return default_value.value();
```